### PR TITLE
Add proto validation for Projects

### PR DESCRIPTION
**Pull request by rdimitrov**
# Summary

Ref https://github.com/stacklok/minder-stories/issues/94

## Change Type

***Mark the type of change your PR introduces:***

- [x] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: proto/minder/v1/minder.proto [hunk 1]
```diff
@@ -2317,7 +2317,14 @@ message UpdateProjectRequest {
     ];
 
     // description is the description of the project to update.
```
Comment (evankanderson) [inline proto/minder/v1/minder.proto:2319]: This is uninterpreted text, right (not Markdown)?

File: proto/minder/v1/minder.proto [hunk 2]
```diff
@@ -2336,7 +2343,14 @@ message ProjectPatch {
...
```
Comment (evankanderson) [inline proto/minder/v1/minder.proto:2345]: Ditto.

File: proto/minder/v1/minder.proto [hunk 3]
```diff
@@ -2317,7 +2317,14 @@ message UpdateProjectRequest {
...
-    string description = 3;
+    string description = 3 [
+        (buf.validate.field).string = {
+            pattern: "[[:space:][:punct:][:word:]]*",
+            min_len: 1,
```
Comment (evankanderson) [inline proto/minder/v1/minder.proto:2323]: If `min_len` is 0, you don't need the `ignore` annotation.

File: proto/minder/v1/minder.proto [hunk 4]
```diff
@@ -2472,9 +2499,18 @@ message Role {
...
+        (buf.validate.field).string = {
+            pattern: "^[a-z]+(_[a-z]+)*$",
+            min_len: 1,
+            max_len: 200,
+        }
+    ];
     // subject is the subject to which the role is assigned.
-    string subject = 2;
+    string subject = 2 [
+        (buf.validate.field).string = {uuid: true},
```
Comment (evankanderson) [inline proto/minder/v1/minder.proto:2511]: I'm not sure these are always UUIDs, but they may happen to be UUIDs for now.  I think it's okay to leave this as-is at the moment.

File: proto/minder/v1/minder.proto (hunk 1 above)
Comment (rdimitrov) [reply proto/minder/v1/minder.proto:2319]: Yes, also confirmed it with Giuseppe 👍

File: proto/minder/v1/minder.proto (hunk 2 above)
Comment (rdimitrov) [reply proto/minder/v1/minder.proto:2379]: Fixed it 👍

File: proto/minder/v1/minder.proto (hunk 3 above)
Comment (rdimitrov) [reply proto/minder/v1/minder.proto]: Fixed it 👍

**Review by evankanderson — APPROVED** [review]

**Pull request merged**